obd/obd.py

```diff
@@ -128,9 +129,11 @@ def __load_commands(self):
             # loop through PIDs bit-array
             for i, bit in enumerate(response.value):
                 if bit:
-
                     mode = get.mode
-                    pid = get.pid + i + 1
+                    if mode == 9:
+                        pid = get.pid + (i - 7) + 1
```

Comment:
Can you comment why mode 9 is different here?

Reply:
Unfortunately, no. From the specification itself, the encoding for mode 9 vs mode 1 PIDs should be the same - however, in the cases where I was able to test, there was an off-by-7 on the PID calculation on mode 9 PIDs compared to mode 1. I'm not convinced that this isn't just a bug in how the result is saved off and so the bitmap itself is skewed, but I have not had the opportunity to dig into this more.

Reply:
Hmm... That isn't ideal.

That comment should at least be in the code if it doesn't match the spec. If it is a bitmap bug then when that is fixed this can be removed.

Reply:
Although thinking about this more we should figure out what is going on here and fix it. I don't really want to add code that expects that the car's responses don't match the spec without good justification. Especially as this looks like it would not work with responses that do match the spec.

It's possible your simulator has a bug in it. I will try to test this series with my set-up and see how it goes.